crates/sui-package-resolver/src/lib.rs

```diff
@@ -308,6 +314,63 @@ impl<S: PackageStore> Resolver<S> {
     }
 }
 
+#[async_trait]
+impl<S: PackageStore> AsyncLayoutResolver for Resolver<S> {
+    async fn get_annotated_layout(
+        &self,
+        struct_tag: &StructTag,
+    ) -> std::result::Result<MoveStructLayout, SuiError> {
+        let move_type_layout = self
+            .type_layout(TypeTag::Struct(Box::new(struct_tag.clone())))
+            .await
+            .map_err(|_e| SuiError::FailObjectLayout {
+                st: format!("{}", struct_tag),
+            })?;
+        let move_struct_layout = match move_type_layout {
+            MoveTypeLayout::Struct(s) => Ok(s),
+            _ => Err(SuiError::FailObjectLayout {
+                st: format!("{}", struct_tag),
+            }),
+        }?;
+        Ok(move_struct_layout)
+    }
+}
+
+#[async_trait]
+impl<S: PackageStore> FunctionSignatureTypesResolver for Resolver<S> {
+    async fn get_function_signature_types(
+        &self,
+        module_id: ModuleId,
+        function: &IdentStr,
+    ) -> Option<Vec<Option<MoveTypeLayout>>> {
+        let pkg_addr = module_id.address();
+        let pkg = self.package_store.fetch(*pkg_addr).await;
+        if let Ok(pkg) = pkg {
```

Comment:
nit: early return using `let ... else`.

I also see that this is the only place that we're currently using the `primitive_type` function from the `sui-json` crate. I'd like to avoid that dependency if we can and I also don't fully see why we need it here. It's doing something quite similar to what the the resolver already does, but limited to primitive types. Is the restriction to primitive types something we need/want, or was it previously a function of not being able to do type resolution?

If the latter, it would be better to add a function to `Resolver` for function signatures.

Reply:
tbh I am not so sure the expected hierarchy of these crates, I just copied the impl of `fn get_signature_types` in `sui-json-rpc-types`, but sure I can copy the impl. from `sui-json` here to avoid the dependency.